src/parseoptions.ts

```diff
@@ -157,13 +160,15 @@ export function parseOptions (options: Partial<Options>) {
       if (isNumber(wday)) {
         byweekday.push(wday)
         continue
+      } else if (isWeekdayStr(wday)) {
+        byweekday.push(Weekday.fromStr(wday).weekday)
+        continue
       }
 
-      const wd = wday as Weekday
```

Comment:
this cast was hiding the bug, and is now not necessary since we've handled all the other cases for the type of `wday`